compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5771,6 +5783,67 @@ public void visit(BLangStringTemplateLiteral stringTemplateLiteral, AnalyzerData
         data.resultType = types.checkType(stringTemplateLiteral, symTable.stringType, data.expType);
     }
 
+    @Override
+    public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral, AnalyzerData data) {
+        //checkStringTemplateExprs(regExpTemplateLiteral.patternFragments, data);
+        data.resultType = types.checkType(regExpTemplateLiteral, symTable.regExpType, data.expType);
+    }
+
+    @Override
```

Comment:
We don't actually need these visitors. These nodes will only be visited at the mapping visitor. The top level regex-literal check would be enough